cfe_internal/core/watchdog/templates/watchdog.mustache

```diff
@@ -1,8 +1,320 @@
+      echo "- ${count} concurrently running cf-agent processes" >> ${COLLECTION_REPORT}
+      PATHOLOGY_COUNT=$((${PATHOLOGY_COUNT}+1))
+    fi
+fi
+
+# Pathology #5: cf-check has encountered a critical issue
+# This indicates that there are one or more integrity issues
+if [ -x /var/cfengine/bin/cf-check ]; then
+    observation=$(/var/cfengine/bin/cf-check diagnose /var/cfengine/state/*.lmdb)
+    if [ $? -ne 0 ]; then
+      echo "$(date) cf-check observed critical integrity issues" >> ${LOGFILE}
+      echo "- cf-check observed critical integrity issues" >> ${COLLECTION_REPORT}
+      echo "  #+begin_example\n${observation}\n#+end_example" >> ${COLLECTION_REPORT}
+    fi
+fi
+
+if [ "${PATHOLOGY_COUNT}" -gt "${PATHOLOGY_THRESHOLD}" ]; then
+
+    echo "$(date) Found ${PATHOLOGY_COUNT} symptoms, threshold (${PATHOLOGY_THRESHOLD}) breached." | tee -a ${LOGFILE}
+    echo "* Observations" >> ${COLLECTION_REPORT}
+
+    ps auxwww > "${COLLECTION_DIR}/ps_auxwww.txt"
+    echo "- [[./ps_auxwww.txt][~ps auxwww~]]" >> ${COLLECTION_REPORT}
+
+    ps -elf > "${COLLECTION_DIR}/ps_-elf.txt"
+    echo "- [[./ps_-elf.txt][~ps -elf~]]" >> ${COLLECTION_REPORT}
+
+    find "${CFENGINE_WORKDIR}" >> "${COLLECTION_DIR}/find__var_cfengine.txt"
+    echo "- [[./find__var_cfengine.txt][=/var/cfengine= file list]] - Before remediation" >> ${COLLECTION_REPORT}
+
+    tar -c -f "${COLLECTION_DIR}/sys.workdir-before-remediation.tar" -C "${CFENGINE_WORKDIR}" state outputs
+    echo "- [[./sys.workdir-before-remediation.tar][=sys.workdir-before-remediation.tar=]] - CFEngine WORKDIR artifacts before remediation" >> ${COLLECTION_REPORT}
+
+    ps -efl | grep cf- > "${COLLECTION_DIR}/cf-procs.txt"
+    echo "- [[./cf-procs.txt][~ps -efl | grep cf-~]]" >> ${COLLECTION_REPORT}
+
+    echo "$(date) Initiating apoptosis" | tee -a ${LOGFILE}
+    while IFS= read -r proc
+    do
+        _PID=$(echo $proc | awk '{print $4}')
+        _PROC=$(echo $proc | awk '{print $15}')
+        _PROCFILE=$(echo "$_PROC" | sed 's./._.g')
+        _COREFILE=$(printf "%s_%s.core" "$_PID" "$_PROCFILE")
+        _COREPATH=$(printf "%s/%s" "${COLLECTION_DIR}" "$_COREFILE")
+        _DBX_ERR_LOG=$(printf "%s/%s.dbx.err" "${COLLECTION_DIR}" "$_COREFILE")
+
+        gencore "$_PID" "$_COREPATH" > "${COLLECTION_DIR}/gencore-$_PID.output" 2>&1
+        echo "- [[./gencore-$_PID.output][output from gencore $_PID $_COREPATH]]" >> ${COLLECTION_REPORT}
+        echo "- [[./$_COREFILE][core from $_PID]]" >> ${COLLECTION_REPORT}
+
+        observation=$(echo "where" | dbx "$_PROC" "$_COREPATH" 2> "$_DBX_ERR_LOG")
+        echo "  - backtrace:" >> ${COLLECTION_REPORT}
+        echo "    #+begin_example\n${observation}\n#+end_example" >> ${COLLECTION_REPORT}
+        echo "  - dbx stderr: [[./$(basename $_DBX_ERR_LOG)][dbx $_PROC $_COREPATH]]" >> ${COLLECTION_REPORT}
+        kill -9 "$_PID" > "${COLLECTION_DIR}/kill_$_PID.txt" 2>&1
```

Comment:
please use signal name instead of number

Reply:
ack